### PR TITLE
test(storage): cleanup options for aggregate download benchmark

### PR DESCRIPTION
**Pull request by coryan**
Use `google::cloud::Options` directly. Simplifies the initialization and
the parsing code.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9651)

### CODE REVIEW

File: google/cloud/storage/benchmarks/aggregate_download_throughput_benchmark.cc [hunk 1]
```diff
@@ -120,21 +119,17 @@ class Iteration {
 };
 
 gcs::Client MakeClient(AggregateDownloadThroughputOptions const& options) {
-  auto client_options =
-      google::cloud::Options{}
-          .set<gcs_experimental::HttpVersionOption>(options.rest_http_version)
-          .set<gcs::TransferStallTimeoutOption>(options.download_stall_timeout);
+  auto opts = google::cloud::Options{options.client_options}
+                  // Make the upload buffer size small, the library will flush
```
Comment (devbww) [inline google/cloud/storage/benchmarks/aggregate_download_throughput_benchmark.cc:123]: s/small, the/small so the/

File: google/cloud/storage/benchmarks/aggregate_download_throughput_options.cc [hunk 2]
```diff
@@ -63,30 +107,51 @@ ParseAggregateDownloadThroughputOptions(std::vector<std::string> const& argv,
...
+       }},
+      {"--grpc-endpoint", "change the gRPC endpoint",
+       [&options](std::string const& val) {
+         options.client_options.set<EndpointOption>(val);
        }},
       {"--download-stall-timeout",
-       "abort downloads stalled for more than this time",
+       "configure `storage::DownloadStallTimeoutOption`: the maximum time"
+       " allowed for data to 'stall' (make insufficient progress) on downloads."
+       " This option is intended for troubleshooting, most of the time the"
```
Comment (devbww) [inline google/cloud/storage/benchmarks/aggregate_download_throughput_options.cc:136]: s/troubleshooting, most/troubleshooting. Most/

**Review by devbww — APPROVED** [review]

File: google/cloud/storage/benchmarks/aggregate_download_throughput_benchmark.cc (hunk 1 above)
Comment (dbolduc) [reply google/cloud/storage/benchmarks/aggregate_download_throughput_benchmark.cc]: is setting `UploadBufferSizeOption` in the download benchmark an intentional change?

Comment (coryan) [reply google/cloud/storage/benchmarks/aggregate_download_throughput_benchmark.cc]: Nope. Fixed.

**Pull request merged**